### PR TITLE
CI: update go workflow

### LINKED ISSUE #38: Add a GH action to lint the codebase with golangci-lint

### PR DESCRIPTION
**Pull request by ctlong**
Closes #38

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - cron: '00 16 * * 1'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
```
Comment (acrmp) [inline .github/workflows/go.yml:18]: Should we consider [getting the go version from the go mod instead](https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file)?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  # Computes and checks the cyclomatic complexity of functions.
+  - gocyclo
+  # Inspects source code for security problems.
+  - gosec
+
+linters-settings:
+  gocyclo:
+    # Minimal code complexity to report.
+    # https://github.com/fzipp/gocyclo#readme
+    min-complexity: 20
```
Comment (acrmp) [inline .golangci.yml:15]: Are we bumping the `min-complexity` from the default?

File: .github/workflows/go.yml (hunk 1 above)
Comment (ctlong) [reply .github/workflows/go.yml]: Good call!

File: .golangci.yml (hunk 2 above)
Comment (ctlong) [reply .golangci.yml:15]: Yes, we are. [The default is 30, but 10-20 is recommended](https://github.com/golangci/golangci-lint/blob/master/.golangci.reference.yml#L555-L558).

Comment (ctlong) [reply .golangci.yml:15]: I've pushed a commit to document that.

**Review by Benjamintf1 — APPROVED** [review]

**Pull request merged**